torch/csrc/jit/tensorexpr/stmt.h

```diff
@@ -16,6 +17,17 @@ class Stmt : public KernelScopedObject {
   Stmt() {}
   TORCH_API virtual void accept(IRVisitor* visitor) const = 0;
   virtual Stmt* accept_mutator(IRMutator* mutator) = 0;
+
+  Stmt* get_parent() const {
+    return parent_;
+  }
+
+ protected:
+  static void set_parent(Stmt* s, Stmt* new_parent) {
+    s->parent_ = new_parent;
+  }
+ private:
+  Stmt* parent_ = nullptr;
```

Comment:
Given how easy it is to miss one to update the parent_ link, it is a good idea to introduce a parent_ integrity checker. At least enable it for tests.